### PR TITLE
feat(vue): update to vite 4 and remove express

### PR DESCRIPTION
**Pull request by louis-bompart**
## Proposed changes

 - Replace the 'mix+express' sauce by a bona fide vite plugin.
 - Refactor env check between the frontend and backend
 - Refresh some tsconfig
 - Dependency cleaning

## Breaking changes

None: The behavior is strictly identical.

## Testing

- Existing FT.
- Manual testing

### CODE REVIEW

**Comment by louis-bompart** [comment]
Eventually, if we move the other framework to vite too, we'll probably replace the search-token-server and search-token-lambda with a vite-plugin dedicated for the search token stuff, so that we DRY

File: packages/ui/vue/create-headless-vue/template/src/commons/isEnvValid.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import z from "zod";
+import { isBrowser } from "browser-or-node";
+
+const zValidEnvironment = z.object({
+  VITE_COVEO_PLATFORM_URL: z.optional(z.string()),
+  VITE_COVEO_ORGANIZATION_ID: z.string(),
+  SERVER_COVEO_API_KEY: isBrowser ? z.undefined() : z.string(),
```
Comment (louis-bompart) [inline packages/ui/vue/create-headless-vue/template/src/commons/isEnvValid.ts:7]: Another addition: Tableflip if you send the API Key in the browser :suspect:

**Review by olamothe — APPROVED** [review]

File: packages/ui/vue/create-headless-vue/template/README.md [hunk 2]
```diff
@@ -7,10 +7,10 @@ This template should help get you started developing your Coveo Search Page with
...
 If you didn't use the Coveo CLI, copy `.example.env` to `.env` and fill it out with the proper information:
 
-- `VITE_APP_PLATFORM_URL`: the URL you usually use to use the Coveo APIs, usually `https://platform.cloud.coveo.com`
-- `VITE_APP_ORGANIZATION_ID`: Your Coveo Organization ID. Check the doc to see [how to retrieve your organization ID](https://docs.coveo.com/en/148/manage-an-organization/retrieve-the-organization-id)
-- `VITE_APP_API_KEY`: Your API Key for this Org. We recommend the 'Search' API Key preset.
-- `VITE_APP_USER_EMAIL`: Your email address. Used for security identities and Analytics.
+- `VITE_COVEO_PLATFORM_URL`: the URL you usually use to use the Coveo APIs, usually `https://platform.cloud.coveo.com`
+- `VITE_COVEO_ORGANIZATION_ID`: Your Coveo Organization ID. Check the doc to see [how to retrieve your organization ID](https://docs.coveo.com/en/148/manage-an-organization/retrieve-the-organization-id)
+- `PRIVATE_APP_API_KEY`: Your API Key for this Org. We recommend the 'Search' API Key preset.
```
Comment (y-lakhdar) [inline packages/ui/vue/create-headless-vue/template/README.md:12]: A bit late but did you mean `SERVER_COVEO_API_KEY`

Comment (louis-bompart) [reply packages/ui/vue/create-headless-vue/template/README.md:12]: Ah yeah good catch

Comment (louis-bompart) [reply packages/ui/vue/create-headless-vue/template/README.md:12]: #1089

**Pull request merged**